### PR TITLE
updates to mosek interface

### PR DESCRIPTION
**Pull request by rileyjmurray**
This pull request contains updates to resolve issue #440 . The effective changes are (1) dual variables are not recovered for discrete problems, and (2) the user can specify if they want a basic feasible solution, if the model is in fact an LP. As a result of these changes, mosek_qpif was removed from cvxpy, and mosek_conif was added to the set of modules covered by test_qp. 

Mosek is currently not passing some of the tests in test_constant_atoms that involve the exponential cone; this is likely because my version of mosek 9 is only an alpha release. For the time being, I have added the offending atoms to a list of exclusions in test_constant_atoms, so that I could run the entire set of nosetests (note: no one will see the effects of this change unless they have mosek 9 installed on their machine). With these exclusions, all nosetests are passing.

### CODE REVIEW

File: cvxpy/tests/test_constant_atoms.py [hunk 1]
```diff
@@ -286,7 +290,7 @@ def run_atom(atom, problem, obj_val, solver, verbose=False):
         print(problem.objective)
         print(problem.constraints)
         print("solver", solver)
-    if check_solver(problem, solver):
+    if check_solver(problem, solver) and not (atom, solver) in KNOWN_SOLVER_ERRORS:
```
Comment (SteveDiamond) [inline cvxpy/tests/test_constant_atoms.py]: why did you change this? It might get lucky and work :)

Comment (rileyjmurray) [reply cvxpy/tests/test_constant_atoms.py]: There was already a check like 

'try:
   problem.solve(solver=solver)
catch:
    assert( (atom, solver) in KNOWN_SOLVER_ERRORS )'

This works when a solver's error is catch-able by Python. But for the alpha release of MOSEK 9 I currently have, some error codes kill the entire Python thread (and so are not catch-able). The only way for me to run all tests in test_constant_atoms.py was to preemptively avoid creating such fatal errors.

In principle, the changes to this file could be reverted. They only apply to people with installations of mosek 9, and hopefully these issues will be fixed well before mosek 9 is released to the public.

**Comment by SteveDiamond** [comment]
Travis CI is failing. You need to have MOSEK only called if it's installed. Fix that and it looks good to merge.

**Comment by rileyjmurray** [comment]
@SteveDiamond 

Can you check the Travis CI logs? Three of the six configurations pass. Of those that fail, none of them reference files that I changed (and some of the issues are timing errors, when warm starting fails to speed up the solver).

**Comment by SteveDiamond** [comment]
Ok when I looked before there were errors were due to your changes.
